### PR TITLE
notebooks now save email information along with notebook info

### PR DESCRIPTION
**Pull request by ksshannon**

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -593,6 +593,19 @@ func (srv *notebookServer) newNotebookHandler(w http.ResponseWriter, r *http.Req
 		return
 	}
 
+	email := ""
```
Comment (ksshannon) [inline server.go]: Zero value for a string is "", this isn't needed.

File: server.go [hunk 2]
```diff
@@ -634,6 +647,30 @@ func (srv *notebookServer) newNotebookHandler(w http.ResponseWriter, r *http.Req
 	handlerPath := path.Join("/book", tmpnb.hash) + "/"
 	log.Printf("handler: %s", handlerPath)
 	handler := func(w http.ResponseWriter, r *http.Request) {
+		//BUG If a user provides a fictitious cookie that could
```
Comment (ksshannon) [inline server.go]: Your code handles this now, correct?  No need to document it.

File: server.go [hunk 3]
```diff
@@ -634,6 +647,30 @@ func (srv *notebookServer) newNotebookHandler(w http.ResponseWriter, r *http.Req
...
+		//represent another user, they could theoretically gain access
+		//to a notebook which does not belong to them.
+
+		//Read the cookie for session information and compare the
+		//email to the email provided by the tmpnb. If they match,
+		//allow access, else redirect them to /list
+		e := ""
```
Comment (ksshannon) [inline server.go]: See above, also maybe `email` would be better, to be consistent.

File: server.go [hunk 4]
```diff
@@ -634,6 +647,30 @@ func (srv *notebookServer) newNotebookHandler(w http.ResponseWriter, r *http.Req
 	handlerPath := path.Join("/book", tmpnb.hash) + "/"
 	log.Printf("handler: %s", handlerPath)
 	handler := func(w http.ResponseWriter, r *http.Request) {
+		//BUG If a user provides a fictitious cookie that could
+		//represent another user, they could theoretically gain access
+		//to a notebook which does not belong to them.
+
+		//Read the cookie for session information and compare the
```
Comment (ksshannon) [inline server.go]: nit: space after `//`

File: server.go (hunk 1 above)
Comment (ksshannon) [reply server.go]: Nevermind, these are fine...

File: server.go (hunk 3 above)
Comment (Erog38) [reply server.go]: email is already a variable in scope from the lines above. I just declared another var for simplicity

File: server.go (hunk 4 above)
Comment (Erog38) [reply server.go]: :+1:

**Pull request merged**